src/main/java/school/hei/haapi/endpoint/rest/mapper/AwardedCourseMapper.java

```diff
@@ -61,4 +70,19 @@ public List<school.hei.haapi.model.AwardedCourse> getAwardedCoursesDomainByGroup
     }
     return awardedCourses;
   }
+
+  public school.hei.haapi.model.AwardedCourse fromCreateAwardedCourseToAwardedCourse(
+      CreateAwardedCourse createAwardedCourse) {
+    Group group = groupService.findById(createAwardedCourse.getGroupId());
+    Course course = courseService.getById(createAwardedCourse.getCourseId());
+    User teacher = userService.findById(createAwardedCourse.getMainTeacherId());
+
+    return school.hei.haapi.model.AwardedCourse.builder()
+        .id(createAwardedCourse.getId())
+        .group(group)
+        .course(course)
+        .mainTeacher(teacher)
+        .creationDatetime(Instant.now())
```

Comment:
this means creationDatetime will be updated everytime we update the awarded course which breaks its contract.
Make the creationDatetime not updatable and (annotated with @CreationTimestamp  or use SQL triggers)